x-pack/platform/plugins/shared/alerting/server/rules_client/lib/validate_actions.ts

```diff
@@ -97,6 +97,21 @@ export async function validateActions(
     );
   }
 
+  const workflowsConnectorTypeIds = new Set(
+    allConnectorTypes.filter((type) => type.subFeature === 'workflows').map((type) => type.id)
+  );
+  const workflowsActionTypeIds = actionResults
+    .map((result) => result.actionTypeId)
+    .filter((id) => workflowsConnectorTypeIds.has(id));
+
+  if (workflowsActionTypeIds.length > 0) {
+    errors.push(
+      i18n.translate('xpack.alerting.rulesClient.validateActions.workflowsConnector', {
+        defaultMessage: 'This type of connector cannot be used as alerting actions',
+      })
+    );
+  }
```

Comment:
This approach of checking compatibility looks very narrow. What happens with connectors that only have `cases`, `genAI`...

Why don't we check that the connectors received have the correct `supportedFeatureIds` (I assume it's `alerting` and not sure if `siem` as well), so we cover all scenarios?

Reply:
Discussed offline and addressed in [d0d3861](https://github.com/elastic/kibana/pull/244119/commits/d0d38614d3ffb374773fd0e952f7bd87b6e3c3f2)